### PR TITLE
`Development`: Adjust client typing documentation

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

adds two points to typing documentation for the client docs:
- strict typing
- prefer interface over type

### Steps for Testing

- open https://artemis-platform--7982.org.readthedocs.build/en/7982/dev/guidelines/client.html
- verify that two points have been added to the type chapter

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

image

## Summary by CodeRabbit

- **Documentation**
	- Updated guidelines for preferring interfaces over types for consistency and enforcing strict typing to avoid type errors.

### CODE REVIEW

**Review by georgyia — DISMISSED** [review]
changes are there, lgtm

File: docs/dev/guidelines/client.rst [hunk 1]
```diff
@@ -37,6 +37,8 @@ Some general aspects:
 2. Do not introduce new types/values to the global namespace.
 3. Shared types should be defined in 'types.ts'.
 4. Within a file, type definitions should come first.
+5. Interfaces and types offer almost the same functionality. In order of consistency, choose ``interface`` over ``type`` whenever possible.
+6. Use strict typing to avoid type errors: Do not use ``any``
```
Comment (Strohgelaender) [inline docs/dev/guidelines/client.rst:41]:
```suggestion
6. Use strict typing to avoid type errors: Do not use ``any``.
```

**Review by nityanandaz — APPROVED** [review]
Looks good to me

**Review by Strohgelaender — APPROVED** [review]

**Review by georgyia — DISMISSED** [review]
reapprove

**Review by laurenzfb — APPROVED** [review]

**Review by reschandreas — APPROVED** [review]

**Review by mateusmm01 — APPROVED** [review]
Lgtm

**Review by rstief — APPROVED** [review]
tested with the given link, looks good 👍

**Pull request merged**